validator/rules/config.go

```diff
@@ -0,0 +1,72 @@
+package rules
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/aerogear/charmil/validator"
+	"github.com/imdario/mergo"
+	"github.com/spf13/cobra"
+)
+
+// ValidatorConfig is provided to user for overriding default rules
+type ValidatorConfig struct {
+	ValidatorOptions `json:"ValidatorOptions"`
+	ValidatorRules   `json:"ValidatorRules"`
+}
+type ValidatorOptions struct {
+	Verbose bool `json:"Verbose"`
+}
+type ValidatorRules struct {
+	Length    `json:"Length"`
+	MustExist `json:"MustExist"`
+}
+
+// ExecuteRules executes all the rules
+// provided by validatorConfig
+func (validatorConfig *ValidatorConfig) ExecuteRules(cmd *cobra.Command) []validator.ValidationError {
+	var ruleConfig RuleConfig
+	return ruleConfig.ExecuteRulesInternal(cmd, validatorConfig)
+}
+
+func ValidatorConfigToRuleConfig(validatorConfig *ValidatorConfig, ruleConfig *RuleConfig) {
+	defaultVerbose := validatorConfig.ValidatorOptions.Verbose
+
+	defaultConfigJson := `{
+		"ValidatorOptions": {
+			"Verbose": false
+		},
+		"ValidatorRules": {
+			"Length": {
+				"Limits": {
+					"Use":     {"Min": 2},
+					"Short":   {"Min": 15},
+					"Long":    {"Min": 50},
+					"Example": {"Min": 50}
+				}
+			},
+			"MustExist": {
+				"Fields": {"Use": true, "Short": true, "Long": true, "Example": true}
+			}
+		}
+	}`
+
+	// unmarshal defaultConfigJson in configHelper
+	var configHelper ValidatorConfig
+	if err := json.Unmarshal([]byte(defaultConfigJson), &configHelper); err != nil {
+		log.Fatal(err)
+	}
+
+	configHelper.Length.Verbose = defaultVerbose
+	configHelper.MustExist.Verbose = defaultVerbose
+
+	// Merge user provided config into configHelper
+	if err := mergo.Merge(&configHelper, validatorConfig, mergo.WithSliceDeepCopy); err != nil {
```

Comment:
Do we still need to use mergo?

Reply:
yes..
`default = Fields{"Use": true, "Short": true, "Long": true, "Example": true}`
if user specified `Fields: {"Args": true, "Long": false}`, 
then without mergo it will be - `Fields: {"Args": true, "Long": false}`

but it should be  `Fields{"Use": true, "Short": true, "Long": false, "Example": true, "Args": true}`